### ISSUE TITLE
Caching middleware breaks compatibility with Django 1.11 and 2.0

### ISSUE THREAD

**Issue by dblenkus** [django/channels#1175]
Middleware caching introduced in 2.1.4 (https://github.com/django/channels/commit/b0f8d368de89bc3d6dd509c2d169b8504ad86793) supports only attributes from Django 2.1 (https://github.com/django/django/blob/stable/2.1.x/django/core/handlers/base.py#L17-L20).

This breaks compatibility with older versions of Django (1.11 and 2.0) as they require additional attributes (https://github.com/django/django/blob/stable/1.11.x/django/core/handlers/base.py#L28-L33).

Following requests fail with:
```
TypeError: 'NoneType' object is not iterable
  File "django/core/handlers/base.py", line 130, in get_response
    for middleware_method in self._response_middleware:
```

cc @majgaard

**Comment by andrewgodwin** [comment]
Ah, drat. I'll have to roll back the release if we can't fix this.

**Comment by dblenkus** [comment]
Fixing this is pretty straight forward: https://github.com/django/channels/pull/1176

The tricky part is writing test for this. The most reliable would probably be to make two request-response cycles instead of mocked test in the commit referenced above. I can work on this tomorrow or on Wednesday.

**Comment by andrewgodwin** [comment]
OK, this is fixed and released in 2.1.5.
